### ISSUE TITLE
Use LoginHandler on page visit

### ISSUE THREAD

**Issue by timkpaine** [QuantStack/voila#239]
This will make it substantially easier to integrate with existing jupyterlab-style custom login handlers

**Comment by maartenbreddels** [comment]
Could you elaborate on this? What do you mean with page visit for instance?

**Comment by timkpaine** [comment]
@maartenbreddels for our use, when a user goes to our jupyterhub instance, we authenticate and do the equivalent of sudoing to their user account. This makes sure that any notebooks they run, including notebooks shared across multiple users, are run as themselves, making sure they can only access the data they're allowed to. We are also hosting voila instances targeting specific notebooks, and want the same hook to sudo to the users accessing the notebook, so that when the notebook is run, their individual user entitlements are enforced.

CC @vidartf

**Comment by vidartf** [comment]
Just to clarify: This issue about adding the authentication setup that `ServerApp` currently has in `jupyter_server` (https://github.com/jupyter/jupyter_server/blob/b45de6c9264c3a704a01eb549fecef4cc6c2341f/jupyter_server/serverapp.py#L1007-L1019, etc.). 

Related to https://github.com/QuantStack/voila/issues/245 and https://github.com/QuantStack/voila/issues/216 .

**Comment by vidartf** [comment]
@timkpaine I think #270 will resolve this. Are you able to verify?

**Comment by timkpaine** [comment]
@vidartf let me check now

**Comment by Zsailer** [comment]
Hi y'all, let me know if this doesn't work for you. 

Jupyter server should bring all the classic notebook's authentication/login handlers with it allowing Voila to use those handlers to control access to the underlying server. That said, the server is a little behind the classic notebook server. We're working to port PRs from notebook to server in the meantime. If you're blocked on this, let me know.

**Comment by timkpaine** [comment]
Closing in favor of `phoila` and #270

**Comment by SylvainCorlay** [comment]
@timkpaine I am curious about "phoila". Is it your goal to fork voila in the long run?

**Comment by vidartf** [comment]
@SylvainCorlay I would say the goal is to upstream as much as possible, and if anything remains, keep it as a thin layer on top of voila (now there is quite a bit of "pre-emptive" work until consensus is reached on the relevant decisions for `jupyter_server`, and until #270 is merged).

**Comment by maartenbreddels** [comment]
Making a single bundle is something that STScI (cc @nmearl) is also interested in. I might open an issue on that here on the voila repo to describe some use case and see if we can 'backport' phoila's design here.

**Comment by timkpaine** [comment]
@SylvainCorlay yeah what @vidartf said, we just are in a rush ;-)
